### PR TITLE
Prevent WP unslash removing already sanitized input.

### PR DESCRIPTION
**Pull request by JoryHogeveen**
Fixes: #5040
Fixes: #4424

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.
- [x] My code includes PHP Unit Tests (if applicable)

### CODE REVIEW

File: classes/PodsAPI.php [hunk 1]
```diff
@@ -310,6 +310,10 @@ public function save_post_meta( $id, $post_meta = null, $strict = false, $fields
 		}
 
 		foreach ( $post_meta as $meta_key => $meta_value ) {
+
+			// Prevent WP unslash removing already sanitized input.
```
Comment (sc0ttkclark) [inline classes/PodsAPI.php:314]: It's possible some of this data is already sanitized/slashed when it gets to here via `PodsAPI::save_post()` which gets called from `PodsAPI::save_object()`

Comment (JoryHogeveen) [reply classes/PodsAPI.php:314]: That is the whole point.
WP core unslashes all data that comes through so we need to double-slash it to make suure the "intended" slashes preserve.

**Review by sc0ttkclark — APPROVED** [review]

**Comment by JoryHogeveen** [comment]
Weird, the tests worked locally, I'll fix the tests and mark the PR ready after that.

File: classes/PodsAPI.php [hunk 2]
```diff
@@ -155,12 +155,16 @@ public function save_wp_object( $object_type, $data, $meta = array(), $strict =
 			$object_type = 'term';
 		}
 
+		$is_meta_object = in_array( $object_type, array( 'post', 'term', 'user', 'comment' ) );
```
Comment (tr1b0t) [inline classes/PodsAPI.php:158]: :no_entry_sign: Short array syntax must be used to define arrays

<sub>hash: 5149bb30bfe969a31482016b18c1d9bc</sub>

File: classes/PodsAPI.php [hunk 3]
```diff
@@ -155,12 +155,16 @@ public function save_wp_object( $object_type, $data, $meta = array(), $strict =
...
+
 		if ( $sanitized ) {
 			$data = pods_unsanitize( $data );
-			$meta = pods_unsanitize( $meta );
+
+			// Do not unsanitize metadata. This is already done by WP.
+			//$meta = pods_unsanitize( $meta );
```
Comment (tr1b0t) [inline classes/PodsAPI.php:164]: * :no_entry_sign: No space found before comment text; expected "// $meta = pods_unsanitize( $meta );" but found "//$meta = pods_unsanitize( $meta );"
* :no_entry_sign: Inline comments must end in full-stops, exclamation marks, or question marks

<sub>hash: 3542bce908033fa8f0998ecd7f0a2085</sub>

**Review by tr1b0t — COMMENTED** [review]
:no_entry_sign: 3 error(s)

File: classes/PodsAPI.php [hunk 4]
```diff
@@ -155,12 +155,14 @@ public function save_wp_object( $object_type, $data, $meta = array(), $strict =
 			$object_type = 'term';
 		}
 
+		$is_meta_object = in_array( $object_type, array( 'post', 'term', 'user', 'comment' ) );
```
Comment (sc0ttkclark) [inline classes/PodsAPI.php:158]:
```suggestion
		$is_meta_object = in_array( $object_type, array( 'post', 'term', 'user', 'comment' ), true );
```

File: tests/phpunit/includes/Meta/Test_Metadata.php [hunk 5]
```diff
@@ -46,6 +44,9 @@ public static function wpSetUpBeforeClass() {
 
 	public static function wpTearDownAfterClass() {
 		foreach ( self::$pod_names as $type => $name ) {
+			if ( in_array( $name, array( 'comment', 'user' ) ) ){
```
Comment (sc0ttkclark) [inline tests/phpunit/includes/Meta/Test_Metadata.php:47]:
```suggestion
			if ( in_array( $name, array( 'comment', 'user' ), true ) ){
```

File: tests/phpunit/includes/Meta/Test_Metadata.php [hunk 6]
```diff
@@ -56,57 +57,55 @@ public static function create_pods() {
 
 		foreach ( self::$pod_names as $type => $name ) {
 
-			self::$pod_ids[ $type ] = pods_api()->save_pod(
+			pods_api()->save_pod(
```
Comment (sc0ttkclark) [inline tests/phpunit/includes/Meta/Test_Metadata.php:60]: Saving and passing the pod id is faster for testing purposes, it's not necessary but if we're looking for tests to run quickly and not time out -- it's helpful

**Review by sc0ttkclark — APPROVED** [review]

**Review by sc0ttkclark — DISMISSED** [review]

**Comment by sc0ttkclark** [comment]
Ready for final QA

**Comment by sc0ttkclark** [comment]
@JoryHogeveen should we punt this to 2.8?

**Comment by sc0ttkclark** [comment]
@JoryHogeveen pending your final QA

**Review by sc0ttkclark — APPROVED** [review]

**Comment by JoryHogeveen** [comment]
Couldn't find any issues locally.

**Review by sc0ttkclark — DISMISSED** [review]

**Pull request merged**